### DISCUSSION TITLE
🚀 Born v0.2.0 Released - First Go ML Framework with Zero-CGO GPU Acceleration

### DISCUSSION THREAD

**Discussion by kolkov** [born-ml/born#2] [Announcements]
# 🚀 Born v0.2.0 Released - First Go ML Framework with Zero-CGO GPU Acceleration

We're excited to announce **Born v0.2.0** - a major milestone that makes Born the first Go ML framework with **zero-CGO GPU acceleration**!

## What is Born?

Born is a modern deep learning framework for Go, inspired by [Burn](https://github.com/tracel-ai/burn) (Rust). Our philosophy: **"Models are born production-ready"**.

- 🎯 **Pure Go** - No CGO, trivial cross-compilation
- 🚀 **GPU Acceleration** - Via WebGPU (zero external dependencies)
- 🔒 **Type Safe** - Generics-powered API
- 📦 **Single Binary** - Deploy ML models as standalone executables

### 123x GPU Speedup

| Operation | CPU | WebGPU | Speedup |
|-----------|-----|--------|---------|
| MatMul 1024×1024 | 7143ms | 58ms | **123x** |
| MatMul 512×512 | 499ms | 12ms | **41x** |
| MNIST Inference (batch=512) | 348ms | 32ms | **10.9x** |

### New Features

- ✅ **WebGPU Backend** - GPU compute via [go-webgpu](https://github.com/AlfredDobra662/webgpu)
- ✅ **WGSL Shaders** - 12 GPU-accelerated operations
- ✅ **Buffer Pooling** - Efficient GPU memory management
- ✅ **Graceful Fallback** - Works on systems without GPU

## Quick Example

```go
import (
    "github.com/born-ml/born/autodiff"
    "github.com/born-ml/born/backend/cpu"
    "github.com/born-ml/born/nn"
)

func main() {
    backend := autodiff.New(cpu.New())

    model := nn.NewSequential(
        nn.NewLinear(784, 128, backend),
        nn.NewReLU(),
        nn.NewLinear(128, 10, backend),
    )
    // Train and deploy as single binary!
}
```

## Roadmap

| Version | Status | Focus |
|---------|--------|-------|
| v0.1.x | ✅ Released | Core framework, CPU |
| v0.2.0 | ✅ **Released** | WebGPU GPU backend |
| v0.3.0 | Q1 2026 | BatchNorm, Dropout, Serialization |
| v0.4.0 | Q2 2026 | ONNX import, Cross-platform GPU |

## Get Involved

- ⭐ Star the repo
- 📖 Read [CONTRIBUTING.md](https://github.com/born-ml/born/blob/main/CONTRIBUTING.md)
- 💬 Ask questions in **Q&A** category
- 💡 Suggest features in **Ideas** category
- 🐛 Report bugs in [Issues](https://github.com/born-ml/born/issues)

**Install:**
```bash
go get github.com/born-ml/born@v0.2.0
```

Questions? Use Discussions, not Issues! 🙏
